dashboard/test/controllers/api/v1/ml_models_controller_test.rb

```diff
@@ -3,12 +3,164 @@
 class Api::V1::MlModelsControllerTest < ::ActionController::TestCase
   setup do
     AWS::S3.stubs(:delete_from_bucket).returns(true)
+    AWS::S3.stubs(:upload_to_bucket).returns(true)
     @owner = create :student
-    @model = create :user_ml_model,  user: @owner
+    @model = create :user_ml_model, user: @owner
     @not_owner = create :student
+    @params_ml_model = {
+      "name" => "venom breath",
+      "datasetDetails" => {
+        "description" => "description.",
+        "numRows" => 101
+      },
+      "selectedTrainer" => "knnClassify",
+      "featureNumberKey" => {
+        "Breathes" => {
+          "Yes" => 0,
+          "No" => 1
+        },
+      "Venomous" => {
+        "No" => 0,
+        "Yes" => 1
+      }
+      },
+      "label" => {
+        "id" => "Venomous",
+        "description" => "Is the animal venomous?",
+        "values" => %w[No Yes]
+      },
+      "features" => [{"id" => "Breathes",
+                      "description" => "Does the animal breathe?",
+                      "values" => %w[Yes No]}],
+      "summaryStat" => {"type" => "classification",
+                        "stat" => "90.00"},
+      "trainedModel" => {"name" => "KNN"}
+    }
```

Comment:
Hmmm I have mixed feelings about including all of this metadata. I don't think we need it to test what you're trying to test and it will be rigid and hard to keep up to date (i.e. it's already out of sync with [new metadata fields](https://github.com/code-dot-org/ml-playground/pull/192))